src/dvc_render/vega.py

```diff
@@ -137,3 +197,293 @@ def generate_markdown(self, report_path=None) -> str:
+            split_anchors,
+            optional_anchors,
+            "group_by_x",
+            [*group_by, self.properties.get("x")],
+        )
+        self._fill_optional_anchor(
+            split_anchors,
+            optional_anchors,
+            "group_by_y",
+            [*group_by, self.properties.get("y")],
+        )
+
+    def _fill_tooltip(
+        self,
+        split_anchors: List[str],
+        optional_anchors: List[str],
+        additional_fields: Optional[List[str]] = None,
+    ):
+        if not additional_fields:
+            additional_fields = []
+        self._fill_optional_anchor(
+            split_anchors,
+            optional_anchors,
+            "tooltip",
+            [
+                {"field": REV},
+                {"field": self.properties.get("x")},
+                {"field": self.properties.get("y")},
+                *[{"field": field} for field in additional_fields],
+            ],
+        )
+
+    def _fill_optional_anchor(
+        self,
+        split_anchors: List[str],
+        optional_anchors: List[str],
+        name: str,
+        value: Any,
+    ):
+        if name not in optional_anchors:
+            return
+
+        optional_anchors.remove(name)
+
+        if name in split_anchors:
+            self._set_split_content(name, value)
+            return
+
+        self.template.fill_anchor(name, value)
+
+    def _get_domain(self, varied_keys: List[str], varied_values: Dict[str, set]):
+        if len(varied_keys) == 2:
+            domain = list(varied_values[CONCAT_FIELDS])
+        else:
+            filename_or_field = varied_keys[0]
+            domain = list(varied_values[filename_or_field])
+
+        domain.sort()
+        return domain
+
+    def _fill_optional_anchor_mapping(
+        self,
+        split_anchors: List[str],
+        optional_anchors: List[str],
+        field: str,
+        name: str,
+        domain: List[str],
+    ):  # pylint: disable=too-many-arguments
+        if name not in optional_anchors:
+            return
+
+        optional_anchors.remove(name)
+
+        encoding = self._get_optional_anchor_mapping(field, name, domain)
+
+        if name in split_anchors:
+            self._set_split_content(name, encoding)
+            return
+
+        self.template.fill_anchor(name, encoding)
+
+    def _get_optional_anchor_mapping(
+        self,
+        field: str,
+        name: str,
+        domain: List[str],
+    ):
+        full_range_values: List[Any] = self._optional_anchor_ranges.get(name, [])
+        anchor_range_values = full_range_values.copy()
+
+        anchor_range = []
+        for _ in range(len(domain)):
+            if not anchor_range_values:
+                anchor_range_values = full_range_values.copy()
+            range_value = anchor_range_values.pop(0)
+            anchor_range.append(range_value)
+
+        legend = (
+            {"legend": {"symbolFillColor": "transparent", "symbolStrokeColor": "grey"}}
```

Comment:
A comment would be helpful here. I remember now that it's about the default being to fill in the legend symbols, but I don't think it's obvious from the code.